ast.ts

```diff
@@ -7,6 +7,7 @@ export type Type =
   | {tag: "none"}
   | {tag: "class", name: string}
   | {tag: "either", left: Type, right: Type }
+  | { tag: "list", listitems : Array<Type>}
```

Comment:
I'm confused by this type. Why will lists have a variable number of _types_ of elements, and why is this part of comprehensions, not lists?